File: performance/test-project/native/js/performance/sample/nativeimpl/StableHasher.js
'use strict';

const { NativeHasher } = require('../nativeinterop/NativeHasher.js');

@NativeImplementation("stable")
class StableHasher extends NativeHasher {
    stable_hash(value) {
        return `stable-${String(value).replace(/:/g, '-')}`;
    }
}

module.exports = { StableHasher };

File: performance/test-project/src/native/java/performance/sample/nativeimpl/StableHasher.java
package performance.sample.nativeimpl;

import dev.capylang.NativeImplementation;
import performance.sample.nativeinterop.NativeHasher;

@NativeImplementation(qualifier = "stable")
public final class StableHasher implements NativeHasher {
    @Override
    public String stable_hash(String value) {
        return "stable-" + value.replace(':', '-');
    }
}
